posix/unix.lisp

```diff
@@ -729,7 +729,9 @@ than C's printf) with format string FORMAT and arguments ARGS."
     (%readdir-r dir entry result)
     (if (null-pointer-p (mem-ref result :pointer))
         nil
-        (with-foreign-slots ((name type fileno) entry (:struct dirent))
+        #+sunos
+        (values (foreign-string-to-lisp (foreign-slot-pointer entry '(:struct dirent) 'name)))
+        #-sunos (with-foreign-slots ((name type fileno) entry (:struct dirent))
           (values (foreign-string-to-lisp name) type fileno)))))
```

Comment:
Small refactoring suggestion:

```lisp
(with-foreign-slots ((name #-sunos type #-sunos fileno) entry (:struct dirent))
  (values (foreign-string-to-lisp name) #-sunos type #-sunos fileno)))))
```

Reply:
No problem, just updated.  I had it that way first but for some reason the 4 #-sunos bothered me.

Reply:
yeah, it bothers me too, but for some reason the duplicated code bothers me more. :-)